pkgs/ffigen/lib/src/code_generator/objc_protocol.dart

```diff
@@ -185,13 +186,15 @@ interface class $name extends $protocolBase $impls{
   /// If `\$keepIsolateAlive` is true, this protocol will keep this isolate
   /// alive until it is garbage collected by both Dart and ObjC.
   static $name implement($args) {
-    final builder = $protocolBuilder();
+    final builder = $protocolBuilder(debugName: '$originalName');
     $buildImplementations
     return $name.castFrom(builder.build(keepIsolateAlive: \$keepIsolateAlive));
   }
 
   /// Adds the implementation of the $originalName protocol to an existing
   /// [$protocolBuilder].
+  ///
+  /// Note: You cannot call this method after you have called `builder.build`.
```

Comment:
Can we add a `boolean` and throw in case someone tries this?

Reply:
Yeah, that's what `ObjCProtocolBuilder._built` does.